great_tables/_tbl_data.py

```diff
@@ -295,8 +298,7 @@ def _(
     elif callable(expr):
         # TODO: currently, we call on each string, but we could be calling on
         # pd.DataFrame.columns instead (which would let us use pandas .str methods)
-        col_pos = {k: ii for ii, k in enumerate(list(data.columns))}
-        return [(col, col_pos[col]) for col in data.columns if expr(col)]
+        return [(col, i) for i, col in enumerate(data.columns) if expr(col)]
```

Comment:
@machow It seems we can directly utilize the index from `enumerate` without constructing another `dict` for lookup.